dvc/tree/s3.py

```diff
@@ -54,18 +54,32 @@ def __init__(self, repo, config):
 
         self._append_aws_grants_to_extra_args(config)
 
+        self.key_id = config.get("s3_key_id")
+        self.key_secret = config.get("s3_key_secret")
+
         shared_creds = config.get("credentialpath")
         if shared_creds:
             os.environ.setdefault("AWS_SHARED_CREDENTIALS_FILE", shared_creds)
+        self._validate_config()
+
+    def _validate_config(self):
+        if bool(self.key_id) != bool(self.key_secret):
+            raise DvcException(
+                "Provide either both or none of `s3_key_id` and `s3_key_secret`."
+            )
```

Comment:
Does `boto3.session` require you pass both at the same time? Or will find the other from env/configs if you only give it one of the keys?